Add page content check to admin/dashboard spec

mjankowski opened the pull request.

mjankowski committed:
Pull out system check message stub to private method in admin/dashboard spec

katafrakt commented:
What's the gain here? It makes the spec even harder to follow: first you need to check the `before` callback and then you need to read a private method (which name is not actually too useful in understanding what the return value is).

mjankowski committed:
Imrove the method name?

mjankowski commented:
> What's the gain here? It makes the spec even harder to follow: first you need to check the `before` callback and then you need to read a private method (which name is not actually too useful in understanding what the return value is).

The `before` was already there, but I agree that in this spec file in particular it's not critical to set up that way. I'd probably remove that as well, but I'm trying to keep these changes small/targetted for ease of review.

Also agreed that the method name could be more useful - I'll push an improved name. I think the reason the stub is there at all is that some of the "checks" would be inappropriate to run in test env, so we want to skip the full call.

mjankowski committed:
Verify response body contents with a system check

mjankowski commented:
Rebased, updated the method name a bit to make it clear WHY we need to only return a subset of the checks, and added an assertion about the html page (look for one of the system check messages).
